pkg/ruler/api.go

```diff
@@ -160,7 +160,7 @@ func (a *API) PrometheusRules(w http.ResponseWriter, req *http.Request) {
 	userID, err := tenant.TenantID(ctx)
 	if err != nil || userID == "" {
 		level.Error(logger).Log("msg", "error extracting org id from context", "err", err)
-		respondServerError(logger, w, "no valid org id found")
```

Comment:
Can we add a test for all the routes? 

It looks like the same will happen on all routes where errors failing to extract the `org_id` in `parseRequest` will cause a server error, looks like we'll have to use `errors.Is` to identify the kind of error it is.

Reply:
as suggested, i reworked the PR to consider all routes, and added the new `TestAPIRoutesCorrectlyHandleInvalidOrgID` unit test.

ready for review again.